Math/NumberTheory/Primes/Factorisation/QuadraticSieve.hs

```diff
@@ -32,32 +63,30 @@ quadraticSieve n b t = runST $ do
         -- modulo 2 as an IntSet.
         sievingFunction = \j -> integerToInt (j ^ (2 :: Int) - n)
```

Comment:
Here `(j ^ (2 :: Int) - n)` need not be an integer when `t` and `n` are large.

Reply:
BTW it would be simpler to write `j * j - n`.